File: templates/javascript-typescript-node/test/profile.test.ts
import {describe, expect, it} from 'vitest';

import {toUserResponse, type User} from '../src/profile.js';

describe('toUserResponse', () => {
  it('returns only public profile fields', () => {
    const user: User = {
      id: 'user-123',
      email: 'reader@example.com',
      displayName: 'Reader Example',
      isActive: true,
    };

    expect(toUserResponse(user)).toEqual({
      id: 'user-123',
      email: 'reader@example.com',
      displayName: 'Reader Example',
    });
  });
});

File: templates/go/profile_test.go
package profile

import "testing"

func TestToUserResponseReturnsOnlyPublicProfileFields(t *testing.T) {
	user := User{
		ID:          "user-123",
		Email:       "reader@example.com",
		DisplayName: "Reader Example",
		IsActive:    true,
	}

	want := UserResponse{
		ID:          "user-123",
		Email:       "reader@example.com",
		DisplayName: "Reader Example",
	}

	got := ToUserResponse(user)
	if got != want {
		t.Errorf("ToUserResponse(%+v) = %+v, want %+v", user, got, want)
	}
}
